docs(CONTRIBUTING): improve guidelines (#6878)
EbookFoundation/free-programming-books: docs/CONTRIBUTING.md

- make sure a book is free. Double-check if needed. It helps the admins if you comment in the PR as to why you think the book is free.
- we don't accept files hosted on Google Drive, Dropbox, Mega, Scribd, Issuu and other similar file upload platforms

> - insert your links in alphabetical order. When there are multiple titles beginning with the same letter order them by the second, and so on. If you see a misplaced link, please reorder it and submit a PR

Author: Maybe we can create a separate section for the alphabetical order and write all these things

> - insert your links in alphabetical order. When there are multiple titles beginning with the same letter order them by the second, and so on. If you see a misplaced link, please reorder it and submit a PR

Reviewer: I don't think this sentence is required. Alphabetical order is IMO enough information for the reader, even if they make a mistake the first time.

> - insert your links in alphabetical order. When there are multiple titles beginning with the same letter order them by the second, and so on. If you see a misplaced link, please reorder it and submit a PR

Author: check the new commit

    GOOD: * [A Way-backed Interesting Book](https://web.archive.org/web/20211016123456/http://example.com/) - John Doe (HTML) *(:card_file_box: archived)*
    ```

### Alphabetical order

- When there are multiple titles beginning with the same letter order them by the second, and so on.

> - `one two` comes before `onetwo`

Reviewer: ```suggestion
For example:
- `aa` comes before `ab`
```

Author: this example makes more sense to put it in the line above (146), to reinforce the concept.
`one two` describes another problem, that of spaces

> If you see a misplaced link, please reorder it and submit a PR.

Reviewer 2: The linter won't allow a misplaced entry, maybe replace this with something about checking the linter?

Author: ```suggestion
If you see an misplaced link, check the linter error message to know which lines should be swapped.
```

Can this be okay?

Reviewer 2: except for "an" it's fine

- No invented titles. We try to take titles from the resources themselves; contributors are admonished not to invent titles or use them editorially if this can be avoided. An exception is for older works; if they are primarily of historical interest, a year in parentheses appended to the title helps users know if they are of interest.
- No ALLCAPS titles. Usually title case is appropriate, but when doubt use the capitalization from the source

> - No emoji or "weird" characters. The reading must be uniform.

Reviewer 3: You could say "Not any special characters or emojis are allowed! For the purpose of readability just use the plain text."

> - No emoji or "weird" characters. The reading must be uniform.

Author: With "special characters" could mean characters like `|` or `&`, maybe you could include "(except  `|`, `&`...)"

> - No emoji or "weird" characters. The reading must be uniform.

Reviewer 3: Yeah, it can be like this

> - No emoji or "weird" characters. The reading must be uniform.

Reviewer 2: This is a book in our list: https://10print.org/

The title is "10 PRINT CHR$(205.5+RND(1)); : GOTO 10"

> - No emoji or "weird" characters. The reading must be uniform.

Author: Then we could simply write "No emoji" reserving the option of disallowing certain titles/characters if we found it necessary, outside of ordinary cases.

> - No emoji or "weird" characters. The reading must be uniform.

Reviewer: The use case of weird in terms of character to some could actually mean  different alphabet systems, I suggest this change
```suggestion
- No emoji or nonstandard characters. The reading must be uniform.
```
Although this is still a little bit vague.

> - No emoji or "weird" characters. The reading must be uniform.

Reviewer 2: I don't recall a case where anyone has submitted a listing containing an emoji or other "odd" character, unless you count `|`.  Let's not try to solve problems we don't have.  We frequently have problem with alphabetical order, for example, so maybe we can focus on that aspect?

> - No emoji or "weird" characters. The reading must be uniform.

Author: The "emoji problem" was on #6874 so I decided to add this rule.

Tomorrow I will commit a draft of the new section regarding alphabetical sorting so that we can discuss it.

> - No emoji or "weird" characters. The reading must be uniform.

Author: check the new commit

- No invented titles. We try to take titles from the resources themselves; contributors are admonished not to invent titles or use them editorially if this can be avoided. An exception is for older works; if they are primarily of historical interest, a year in parentheses appended to the title helps users know if they are of interest.
- No ALLCAPS titles. Usually title case is appropriate, but when doubt use the capitalization from the source

> - No emoji.

Reviewer: ```suggestion
- No emojis.
```
